osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/AbstractGetPowerQualityProfileHandler.java

```diff
@@ -463,39 +439,105 @@ private ProfileEntryValueDto makeNumericProfileEntryValueDto(final DataObject da
-                    final DataObject scalerUnitDataObject = scalerUnitResult.get(0).getResultData();
-                    result.add(new ScalerUnitInfo(logicalName, classId, scalerUnitDataObject));
-                } else {
-                    result.add(new ScalerUnitInfo(logicalName, classId, null));
+
+                if (logicalNames.contains(logicalName)) {
+                    selectableCaptureObjects.put(positionInDataObjectsList,
+                            new CaptureObjectDefinitionDto(cosemObjectDefinitionDto.getClassId(),
+                                    new ObisCodeValuesDto(logicalName),
+                                    (byte) cosemObjectDefinitionDto.getAttributeIndex(),
+                                    cosemObjectDefinitionDto.getDataIndex()));
                 }
+                positionInDataObjectsList++;
+            }
+        }
+
+        return selectableCaptureObjects;
+    }
+
+    private List<ScalerUnitInfo> createScalerUnitInfos(final DlmsConnectionManager conn, final DlmsDevice device,
+            final List<GetResult> captureObjects) throws ProtocolAdapterException {
+
+        List<ScalerUnitInfo> scalerUnitInfos = new ArrayList<>();
+
+        // there is always only one GetResult
+        for (final GetResult captureObjectResult : captureObjects) {
+
+            final List<DataObject> dataObjects = captureObjectResult.getResultData().getValue();
+
+            for (DataObject dataObject : dataObjects) {
+                scalerUnitInfos.add(createScalerUnitInfo(conn, device, dataObject));
             }
         }
 
-        return result;
+        return scalerUnitInfos;
+    }
+
+    private ScalerUnitInfo createScalerUnitInfo(final DlmsConnectionManager conn, final DlmsDevice device,
+            final DataObject dataObject) throws ProtocolAdapterException {
+
+        final CosemObjectDefinitionDto cosemObjectDefinitionDto = this.dlmsHelper
+                .readObjectDefinition(dataObject, CAPTURE_OBJECT);
+
+        final int classId = cosemObjectDefinitionDto.getClassId();
+        final String logicalName = cosemObjectDefinitionDto.getLogicalName().toString();
+
+        if (this.hasScalerUnit(classId)) {
+            final AttributeAddress addr = new AttributeAddress(classId, logicalName, SCALER_UNITS_MAP.get(classId));
+            final List<GetResult> scalerUnitResult = this.dlmsHelper
+                    .getAndCheck(conn, device, "retrieve scaler unit for capture object", addr);
+            final DataObject scalerUnitDataObject = scalerUnitResult.get(0).getResultData();
+            return new ScalerUnitInfo(logicalName, classId, scalerUnitDataObject);
+        } else {
+            return new ScalerUnitInfo(logicalName, classId, null);
+        }
+    }
+
+    private static List<String> getLogicalNamesPublicDefinableLoadProfile() {
+
+        return Arrays.asList(OBIS_CODE_CLOCK, NR_VOLTAGE_SAGS_L1, NR_VOLTAGE_SAGS_L2, NR_VOLTAGE_SAGS_L3,
```

Comment:
These look like enums; consider defining them as such.

Reply:
you are right, how lazy of me :-)